### PR TITLE
Updated Guide on Hoppscotch Widgets with Interactive Examples

### PR DESCRIPTION
**Pull request by ganeshpatil386386**
Solves #203 

- Links: Added detailed guidance on creating and using links.
- Buttons: Provided examples and scenarios for effective button integration.
- Embeds: Included a comprehensive guide along with a live demo.
- Deployed a small template on Netlify to showcase the use of the Embed component.

Added detailed guides and scenarios for each widget type for clearer usage instructions.
Need Review - @SanskritiHarmukh @liyasthomas @thetronjohnson

### CODE REVIEW

File: guides/articles/widgets.mdx [hunk 1]
```diff
@@ -0,0 +1,88 @@
+---
+sidebarTitle: Utilizing Widgets
```
Comment (SanskritiHarmukh) [inline guides/articles/widgets.mdx:2]:
```suggestion
sidebarTitle: Uplifting web experience with Hoppscotch Widgets
```

File: guides/articles/widgets.mdx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+title: Enhancing Your App with Hoppscotch Widgets
```
Comment (SanskritiHarmukh) [inline guides/articles/widgets.mdx:3]:
```suggestion
title: Uplifting web experience with Hoppscotch Widgets
```

File: guides/articles/widgets.mdx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+description:
+  Embed Hoppscotch Widgets to create interactive and dynamic elements on your
```
Comment (SanskritiHarmukh) [inline guides/articles/widgets.mdx:5]:
```suggestion
  Create and add interactive and dynamic elements to your website using Hoppscotch Widgets.
```

File: guides/articles/widgets.mdx [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+  website.
```
Comment (SanskritiHarmukh) [inline guides/articles/widgets.mdx:6]:
```suggestion

```

File: guides/articles/widgets.mdx [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+---
+
+Widgets are interactive components that can be embedded in HTML pages to create
+a seamless user experience. Hoppscotch offers three types of widgets: links,
```
Comment (SanskritiHarmukh) [inline guides/articles/widgets.mdx:10]:
```suggestion
user experience. Hoppscotch offers three types of widgets: links,
```

File: guides/articles/widgets.mdx [hunk 6]
```diff
@@ -0,0 +1,88 @@
+---
+sidebarTitle: Utilizing Widgets
+title: Enhancing Your App with Hoppscotch Widgets
+description:
+  Embed Hoppscotch Widgets to create interactive and dynamic elements on your
+  website.
+---
+
+Widgets are interactive components that can be embedded in HTML pages to create
```
Comment (SanskritiHarmukh) [inline guides/articles/widgets.mdx:9]:
```suggestion
Widgets are interactive components that can be embedded in HTML pages to enhance
```

File: guides/articles/widgets.mdx [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+a seamless user experience. Hoppscotch offers three types of widgets: links,
+buttons, and embeds. These widgets allow you to easily integrate dynamic
+elements into your website or project.
+
+## Links
+
+Links are the simplest type of widget. They are just URLs that you can share
```
Comment (SanskritiHarmukh) [inline guides/articles/widgets.mdx:16]:
```suggestion
Links are the simplest way to share your Hoppscotch requests. You can create and share
```

File: guides/articles/widgets.mdx [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+with others. You can create links for your requests in three formats: raw, HTML,
```
Comment (SanskritiHarmukh) [inline guides/articles/widgets.mdx:17]:
```suggestion
them with others in three formats: raw, HTML, and Markdown. 
```

File: guides/articles/widgets.mdx [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+and Markdown.
```
Comment (SanskritiHarmukh) [inline guides/articles/widgets.mdx:18]:
```suggestion
```

File: guides/articles/widgets.mdx [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+
+Use this link to easily share your request with others or include it in
+documentation and blogs.
+
+**Example:** [Run in Hoppscotch](https://hopp.sh/r/r0338F6yoJrJ)
+
+## Buttons
+
+Buttons offer a more advanced widget option and can be embedded in HTML or
```
Comment (SanskritiHarmukh) [inline guides/articles/widgets.mdx:27]:
```suggestion
Buttons take things up a notch. These can be embedded directly into HTML or
```

File: guides/articles/widgets.mdx [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+Markdown files. You can customize how they look and act to fit your needs.
```
Comment (SanskritiHarmukh) [inline guides/articles/widgets.mdx:28]:
```suggestion
Markdown files, and you can customize them to match your site's look and feel.
```

File: guides/articles/widgets.mdx [hunk 12]
```diff
@@ -0,0 +1,88 @@
...
+
+Embed this button in documentation, website or blogs to provide easy access to
```
Comment (SanskritiHarmukh) [inline guides/articles/widgets.mdx:30]:
```suggestion
Add this button in your project's documentation, website or blogs to provide easy access to
```

File: guides/articles/widgets.mdx [hunk 13]
```diff
@@ -0,0 +1,88 @@
...
+your request.
```
Comment (SanskritiHarmukh) [inline guides/articles/widgets.mdx:31]:
```suggestion
your requests with a simple click.
```

File: guides/articles/widgets.mdx [hunk 14]
```diff
@@ -0,0 +1,88 @@
...
+
+**Example:**
+[![Run in Hoppscotch](https://hopp.sh/badge.svg)](https://hopp.sh/r/r0338F6yoJrJ)
+
+## Embeds
+
+Embeds are the most advanced widgets, allowing you to embed a mini version of
```
Comment (SanskritiHarmukh) [inline guides/articles/widgets.mdx:38]:
```suggestion
Embeds are the most advanced widgets, allowing you to integrate a mini version of
```

File: guides/articles/widgets.mdx [hunk 15]
```diff
@@ -0,0 +1,88 @@
...
+Hoppscotch directly into an HTML page. This lets your audience interact directly
```
Comment (SanskritiHarmukh) [inline guides/articles/widgets.mdx:39]:
```suggestion
Hoppscotch right into your HTML page. This lets your audience interact directly
```

File: guides/articles/widgets.mdx [hunk 16]
```diff
@@ -0,0 +1,88 @@
...
+with your API requests. You can also customize how the embed looks and functions
+to match your preferences.
+
+[Embed Example](https://hoppscotch-embed-widget.netlify.app/)
```
Comment (SanskritiHarmukh) [inline guides/articles/widgets.mdx:43]:
```suggestion
**Example:** [See Embed in Action](https://hoppscotch-embed-widget.netlify.app/)
```

File: guides/articles/widgets.mdx [hunk 17]
```diff
@@ -0,0 +1,88 @@
...
+  values before sharing to ensure it works correctly.
+</Warning>
+
+### How to Access Shared Request
+
+Steps to Create a Shared Request:
+
+1. Click the icon next to the **Save** button and select **"Share Request."**
+2. Choose the widget type: **Link**, **Button**, or **Embed**.
+3. Click the **"Create"** button to generate the links.
```
Comment (SanskritiHarmukh) [inline guides/articles/widgets.mdx:68]:
```suggestion
3. Click the **"Create"** button to generate the widget.
```

File: guides/articles/widgets.mdx [hunk 18]
```diff
@@ -0,0 +1,88 @@
...
+
+### How to Manage Shared Request
+
+To view your shared requests:
+
+1. Click on the **"Shared Requests"** icon in the right-hand sidebar.
+2. You will see a list of all your shared requests.
+3. Click on any request to access it.
+4. Right-click or use the three-dot menu on a request to customize the **widget,
+   copy the link,edit, or delete** the shared request.
```
Comment (SanskritiHarmukh) [inline guides/articles/widgets.mdx:88]:
```suggestion
   copy the link, edit, or delete** the shared request.
```

File: guides/articles/widgets.mdx [hunk 19]
```diff
@@ -0,0 +1,88 @@
...
+</Tip>
+
+### How to Manage Shared Request
+
+To view your shared requests:
+
+1. Click on the **"Shared Requests"** icon in the right-hand sidebar.
+2. You will see a list of all your shared requests.
+3. Click on any request to access it.
+4. Right-click or use the three-dot menu on a request to customize the **widget,
```
Comment (SanskritiHarmukh) [inline guides/articles/widgets.mdx:87]:
```suggestion
4. Right-click or use the three-dot menu on a request to **customize the widget,
```

**Review by SanskritiHarmukh — CHANGES_REQUESTED** [review]

**Comment by SanskritiHarmukh** [comment]
@ganeshpatil386386 , could you also add a reference to this guide on the [Index Page](https://docs.hoppscotch.io/guides/articles) of guides, under [Community Articles](https://docs.hoppscotch.io/guides/articles#community-articles), just above the tip?

**Comment by ganeshpatil386386** [comment]
Add suggested changes looking for review @SanskritiHarmukh @thetronjohnson

File: mint.json [hunk 20]
```diff
@@ -199,6 +199,7 @@
         "guides/articles",
         "guides/articles/restful-api-testing-with-hoppscotch",
         "guides/articles/improving-your-api-workflow",
+        "guides/articles/widgets",
```
Comment (SanskritiHarmukh) [inline mint.json:202]: Can you change the name of the page here as `uplifting-web-experience-with-hoppscotch-widgets` and in the file directory as `uplifting-web-experience-with-hoppscotch-widgets.mdx`.

File: guides/articles.mdx [hunk 21]
```diff
@@ -4,20 +4,57 @@ title: Index
...
 
-<Tip>We're looking for more articles! If you have written an article about Hoppscotch, please [open a pull request](https://github.com/hoppscotch/docs) on our documentation repository. We'll be happy to add it to this list.</Tip>
+<CardGroup cols={1}>
+  <Card
+    title='Uplifting Web Experience with Hoppscotch Widgets'
+    iconType='light'
+    icon='circle-arrow-right'
+    href='/guides/articles/widgets'
+  >
+    Learn how to create and add interactive, dynamic elements to your website
```
Comment (SanskritiHarmukh) [inline guides/articles.mdx:51]:
```suggestion
    Widgets are interactive components that can be embedded in HTML pages to enhance user experience...
```

File: guides/articles.mdx [hunk 22]
```diff
@@ -4,20 +4,57 @@ title: Index
...
+    using Hoppscotch Widgets, including links, buttons, and embeds.
```
Comment (SanskritiHarmukh) [inline guides/articles.mdx:52]:
```suggestion
```

**Review by SanskritiHarmukh — CHANGES_REQUESTED** [review]

**Comment by SanskritiHarmukh** [comment]
@ganeshpatil386386 , you can just click 'Commit Suggestion' to implement the changes I requested if you see that option for the reviews above. Also, could you disable any code formatting tools in your editor while making changes to these files?

**Review by SanskritiHarmukh — APPROVED** [review]

**Comment by SanskritiHarmukh** [comment]
I guess the content is still formatted in the Index file. I'll fix that with another PR. Thank you for making the rest of the requested changes, it looks good to me ✅. @thetronjohnson , could you review this PR from your end as well when you have a chance?

**Pull request merged**